src/core_plugins/elasticsearch/lib/health_check.js

```diff
@@ -15,26 +15,25 @@ const NO_INDEX = 'no_index';
 const INITIALIZING = 'initializing';
 const READY = 'ready';
 
-const REQUEST_DELAY = 2500;
-
 module.exports = function (plugin, server) {
   const config = server.config();
-  const adminClient = server.plugins.elasticsearch.adminClient;
-  const dataClient = server.plugins.elasticsearch.dataClient;
+  const callAdminAsKibanaUser = server.plugins.elasticsearch.getCluster('admin').callAsKibanaUser;
+  const callDataAsKibanaUser = server.plugins.elasticsearch.getCluster('data').callAsKibanaUser;
+  const REQUEST_DELAY = config.get('elasticsearch.healthCheck.delay');
 
   plugin.status.yellow('Waiting for Elasticsearch');
-  function waitForPong(client, url) {
-    return client.ping().catch(function (err) {
+  function waitForPong(callAsKibanaUser, url) {
```

Comment:
`callAsKibanaUser`? Yeah, it's being passed as `callAdminAsKibanaUser` which is defined on line 20. I can see how that is confusing. Thoughts on what we can rename it to? `caller`?